foo_ui_columns/drop_down_list_toolbar.h

```diff
@@ -101,13 +122,19 @@ void DropDownListToolbar<ToolbarArgs>::refresh_all_items()
     m_items = ToolbarArgs::get_items();
 
     ComboBox_ResetContent(m_wnd_combo);
-
-    // auto&& crashes the VS 2017 15.6 compiler here
-    for (auto& [id, name] : m_items) {
-        ComboBox_AddString(m_wnd_combo, pfc::stringcvt::string_wide_from_utf8(name.c_str()));
-        const auto cx = uih::get_text_width(dc, name.c_str(), name.size());
-        m_max_item_width = max(m_max_item_width, cx);
+    ComboBox_Enable(m_wnd_combo, !ranges::empty(m_items));
+
+    if (ranges::empty(m_items) && ToolbarArgs::get_items_empty_text() != nullptr) {
+        ComboBox_AddString(m_wnd_combo, pfc::stringcvt::string_wide_from_utf8(ToolbarArgs::get_items_empty_text()));
```

Comment:
Doh, been testing with different layout settings and it never occurred to me...

Reply:
This does highlight an existing problem where it should be calling `get_host()->on_size_limit_change()` if the minimum width changes because the items have changed... But I'll have a look at that separately as it would've been a problem before with some of the other toolbars.